src/js/hooks/useTableCheckboxes.js

```diff
@@ -0,0 +1,62 @@
+import { useCallback, useEffect, useState } from 'react';
+
+const useTableCheckboxes = () => {
+  const [checkedCheckboxes, setCheckedCheckboxes] = useState([]);
+  const [headerCheckboxState, setHeaderCheckboxState] = useState({
+    indeterminate: false,
+    value: false,
+  });
+
+  const selectRow = (identifier) => useCallback(() => {
+    if (checkedCheckboxes.includes(identifier)) {
+      setCheckedCheckboxes(
+        (checked) => checked.filter((checkedProductId) => identifier !== checkedProductId),
+      );
+      return;
+    }
+    setCheckedCheckboxes((checked) => ([...checked, identifier]));
+  }, [checkedCheckboxes]);
+
+  useEffect(() => {
+    const selectedRows = checkedCheckboxes.length;
```

Comment:
I think you mix two namings here - `selectedRows` vs `checkedCheckboxes`. Maybe we could align on one of them?
Moreover - for me it isn't a problem, but `selectedRows` sound like we store the whole rows there, not only the length, so how about adding any `amountOf` prefix?